demos/demo1-chat-heads/Demo.jsx

```diff
@@ -1,59 +1,222 @@
-import React from 'react';
-import {StaggeredMotion, spring, presets} from '../../src/react-motion';
-import range from 'lodash.range';
+import React from "react";
+import {
+  StaggeredMotion,
+  spring,
+  Motion,
+  presets
+} from "../../src/react-motion";
+import range from "lodash.range";
+import Slider from "./Slider";
 
-export default class Demo extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {x: 250, y: 300};
-  };
+const options = [
+  {
+    description: "Constant",
+    value: "constant"
+  },
+  {
+    description: "Linear",
+    value: "linear"
+  },
+  {
+    description: "Inverse Linear",
+    value: "invertedLinear"
+  }
+];
+
+const Demo = React.createClass({
+  getInitialState() {
+    return {
+      x: 250,
+      y: 300,
+      stiffness: 100,
+      damping: 17,
+      precision: 0.01,
+      stiffnessBehaviour: "constant",
+      dampingBehaviour: "constant",
+      headerHeight: 0
+    };
+  },
 
   componentDidMount() {
-    window.addEventListener('mousemove', this.handleMouseMove);
-    window.addEventListener('touchmove', this.handleTouchMove);
-  };
+    window.addEventListener("mousemove", this.handleMouseMove);
+    window.addEventListener("touchmove", this.handleTouchMove);
+    this.setState({ headerHeight: this.divRef.clientHeight });
+  },
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.x !== nextState.x) {
+      return true;
+    }
+    if (this.props.y !== nextProps.y) {
+      return true;
+    }
+    if (this.state.stiffness !== nextState.stiffness) {
+      return true;
+    }
+    if (this.state.damping !== nextState.damping) {
+      return true;
+    }
+    if (this.state.precision !== nextState.damping) {
+      return true;
+    }
+
+    return false;
+  },
 
-  handleMouseMove = ({pageX: x, pageY: y}) => {
-    this.setState({x, y});
-  };
+  handleMouseMove({ pageX: x, pageY: y }) {
+    const height = this.state.headerHeight;
+    const chatHeadRadiusPx = 20;
+    // chat heads avoid header area
+    if (y > height * 3.3 + chatHeadRadiusPx) {
+      this.setState({ x, y: y - height * 3.3 });
+    }
+  },
 
-  handleTouchMove = ({touches}) => {
+  handleTouchMove({ touches }) {
     this.handleMouseMove(touches[0]);
-  };
+  },
+
+  handleStiffnessChange(event) {
+    this.setState({ stiffness: event.target.value });
+  },
+
+  handlePrecisionChange(event) {
+    if (!isNaN(event.target.value)) {
+      this.setState({ precision: event.target.value });
+    }
+  },
 
-  getStyles = (prevStyles) => {
+  handleDampingChange(event) {
+    this.setState({ damping: event.target.value });
+  },
+
+  handleStiffnessParamChange(event) {
+    this.setState({ stiffnessBehaviour: event.target.value });
+  },
+
+  handleDampingParamChange(event) {
```

Comment:
I'd consider rewriting this as this (utilizing all the good things that ES6 gives us): 

`handleDampingParamChange = ({ target: { value } }) => this.setState({ stiffnessBehaviour: value });`

Or just:

`handleDampingParamChange = (e) => this.setState({ stiffnessBehaviour: e.target.value });`